### PR TITLE
Add ecommerce objc snippets

### PR DESCRIPTION
**Pull request by morganchen12**

### CODE REVIEW

File: firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m [hunk 1]
```diff
@@ -33,4 +50,235 @@ - (void)logInAppPurchase {
...
 
+- (void)enhancedEcommerce {
+  // Implementation
+
+  // A pair of jeggings
+  NSMutableDictionary *jeggings = [@{
+    kFIRParameterItemID: @"SKU_123",
+    kFIRParameterItemName: @"jeggings",
+    kFIRParameterItemCategory: @"pants",
+    kFIRParameterItemVariant: @"black",
```
Comment (samtstern) [inline firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m]: But if the jeggings are black ... are they just leggings? 😮

File: firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m [hunk 2]
```diff
@@ -33,4 +50,235 @@ - (void)logInAppPurchase {
...
+    kFIRParameterShippingTier: @"Ground"
+  } mutableCopy];
+
+  // Add items
+  shippingParams[kFIRParameterItems] = @[jeggings];
+
+  // Log added shipping info event
+  [FIRAnalytics logEventWithName:kFIREventAddShippingInfo parameters:shippingParams];
+
+  // Prepare shipping params
```
Comment (htcgh) [inline firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m]:
```suggestion
  // Prepare payment params
```

File: firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m [hunk 3]
```diff
@@ -33,4 +50,235 @@ - (void)logInAppPurchase {
...
+    kFIRParameterCurrency: @"USD",
+    kFIRParameterValue: @14.98,
+    kFIRParameterCoupon: @"SUMMER_FUN",
+    kFIRParameterPaymentType: @"Visa"
+  } mutableCopy];
+
+  // Add items
+  paymentParams[kFIRParameterItems] = @[jeggings];
+
+  // Log added shipping info event
```
Comment (htcgh) [inline firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m]:
```suggestion
  // Log added payment info event
```

File: firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m [hunk 4]
```diff
@@ -33,4 +50,235 @@ - (void)logInAppPurchase {
...
+    kFIRParameterValue: @14.98,
+    kFIRParameterTax: @2.58,
+    kFIRParameterShipping: @5.34,
+    kFIRParameterCoupon: @"SUMMER_FUN"
+  } mutableCopy];
+
+  // Add items
+  purchaseParams[kFIRParameterItems] = @[jeggings];
+
+  // Log added shipping info event
```
Comment (htcgh) [inline firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m]:
```suggestion
  // Log purchase event
```

File: firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m [hunk 5]
```diff
@@ -33,4 +50,235 @@ - (void)logInAppPurchase {
...
+  [FIRAnalytics logEventWithName:kFIREventPurchase parameters:purchaseParams];
+
+  // Prepare purchase params
```
Comment (htcgh) [inline firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m]:
```suggestion
  // Prepare refund params
```

File: firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m [hunk 6]
```diff
@@ -33,4 +50,235 @@ - (void)logInAppPurchase {
...
+  NSDictionary *refundedProduct = @{
+    kFIRParameterItemID: @"SKU_123",
+    kFIRParameterQuantity: @1,
+  };
+
+  // Add items
+  refundParams[kFIRParameterItems] = @[refundedProduct];
+
+  // Log added shipping info event
+  [FIRAnalytics logEventWithName:kFIREventPurchase parameters:refundParams];
```
Comment (htcgh) [inline firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m]:
```suggestion
  // Log refund event
  [FIRAnalytics logEventWithName:kFIREventRefund parameters:refundParams];
```

File: firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m [hunk 7]
```diff
@@ -33,4 +50,235 @@ - (void)logInAppPurchase {
...
+
+  // Log event
+  [FIRAnalytics logEventWithName:kFIREventViewItem parameters:productDetails];
+
+  // Adding/Removing a product from shopping cart
+
+  // Specify order quantity
+  jeggings[kFIRParameterQuantity] = @2;
+
+  // Prepare ecommerce event
```
Comment (htcgh) [inline firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m]:
```suggestion
  // Prepare item detail params
```

File: firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m [hunk 8]
```diff
@@ -33,4 +50,235 @@ - (void)logInAppPurchase {
...
+
+  // Log an event when product is added to cart
+  [FIRAnalytics logEventWithName:kFIREventAddToCart parameters:itemDetails];
+
+  // Specify order quantity
+  jeggings[kFIRParameterQuantity] = @2;
+  boots[kFIRParameterQuantity] = @1;
+
+  // Prepare order parameters
+  NSMutableDictionary *orderParameters = [@{
```
Comment (htcgh) [inline firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m]:
```suggestion
  NSMutableDictionary *orderParams = [@{
```

File: firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m [hunk 9]
```diff
@@ -33,4 +50,235 @@ - (void)logInAppPurchase {
...
+  // Prepare ecommerce parameters
+  NSMutableDictionary *itemList = [@{
+    kFIRParameterItemListID: @"L001",
+    kFIRParameterItemListName: @"Related products",
+  } mutableCopy];
+
+  // Add items array
+  itemList[kFIRParameterItems] = @[jeggings, boots, socks];
+
+  // Log event
```
Comment (htcgh) [inline firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m]:
```suggestion
  // Log view item list event
```

File: firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m [hunk 10]
```diff
@@ -33,4 +50,235 @@ - (void)logInAppPurchase {
...
+  NSMutableDictionary *selectedItem = [@{
+    kFIRParameterItemListID: @"L001",
+    kFIRParameterItemListName: @"Related products",
+  } mutableCopy];
+
+  // Add items array
+  selectedItem[kFIRParameterItems] = @[jeggings];
+
+  // Log event
+  [FIRAnalytics logEventWithName:kFIREventSelectItem parameters:itemList];
```
Comment (htcgh) [inline firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m]:
```suggestion
  // Log select item event
  [FIRAnalytics logEventWithName:kFIREventSelectItem parameters:selectedItem];
```

File: firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m [hunk 11]
```diff
@@ -33,4 +50,235 @@ - (void)logInAppPurchase {
...
+  // Prepare ecommerce parameters
+  NSMutableDictionary *productDetails = [@{
+    kFIRParameterCurrency: @"USD",
+    kFIRParameterValue: @9.99
+  } mutableCopy];
+
+  // Add items array
+  productDetails[kFIRParameterItems] = @[jeggings];
+
+  // Log event
```
Comment (htcgh) [inline firoptions/FiroptionConfiguration/FiroptionConfiguration/AnalyticsHelper.m]:
```suggestion
  // Log view item event
```

**Pull request merged**